cmd/openshift-install/create.go

```diff
@@ -931,3 +937,34 @@ func IsSingleNode() (bool, error) {
 	}
 	return false, nil
 }
+
+// handleInterrupt returns a new context which will be cancelled with a user interrupt.
+// shutdown is an optional shutdown function that will be executed before the context
+// is cancelled.
+func handleInterrupt(signalCtx context.Context, shutdown func()) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	logrus.RegisterExitHandler(cancel)
+	go func() {
+		<-signalCtx.Done()
+		logrus.Warn("Received interrupt signal")
+		if shutdown != nil {
+			shutdown()
+		}
+		cancel()
+	}()
+	return ctx, cancel
+}
+
+func shutdownCAPISystem() {
+	if sys := clusterapi.System(); sys.State() == clusterapi.SystemStateRunning {
+		sys.Teardown()
```

Comment:
This might be racy, in case let's say the system isn't yet in a running state. How about making sure that Teardown() is re-entrant?

Reply:
Ah ok. I think Teardown is indeed re-entrant; will confirm.